src/IssueViz.Security/Taint/TaintStore.cs

```diff
@@ -83,13 +83,18 @@ public void Add(IAnalysisIssueVisualization issueVisualization)
                 throw new ArgumentNullException(nameof(issueVisualization));
             }
 
-            if (analysisInformation == null)
-            {
-                return;
-            }
-
             lock (Locker)
             {
+                if (analysisInformation == null)
+                {
+                    return;
+                }
+
+                if (taintVulnerabilities.Contains(issueVisualization, TaintAnalysisIssueVisualizationByIssueKeyEqualityComparer.Instance))
+                {
+                    return;
```

Comment:
Could the state of the issue have changed, and if so would we care?

Reply:
Clarified with Damien - the answer is no